zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/IPAuthenticationProvider.java

```diff
@@ -81,6 +96,121 @@ private byte[] v4addr2Bytes(String addr) {
         return b;
     }
 
+    /**
+     * Validates an IPv6 address string and converts it into a byte array.
+     *
+     * @param ipv6Addr The IPv6 address string to validate.
+     * @return A byte array representing the IPv6 address if valid, or null if the address
+     * is invalid or cannot be parsed.
+     */
+    public static byte[] v6addr2Bytes(String ipv6Addr) {
+        // Split the address by "::" to handle zero compression, -1 to keep trailing empty strings
+        String[] parts = ipv6Addr.trim().split("::", -1);
```

Comment:
```suggestion
        String[] parts = ipv6Addr.split("::", -1);
```

To be consistent with `v4addr2Bytes`. I would be strange for them to behave different with spaces.